compiler/testData/codegen/box/casts/javaInterop/enhancedNullabilityConstant.kt

```diff
@@ -0,0 +1,24 @@
+// TARGET_BACKEND: JVM
+// IGNORE_BACKEND_FIR: JVM_IR
+// FILE: box.kt
+fun id(x: Long) = x
+
+fun box(): String {
+    val x = X()
+    val y = id(x.y ?: 0 * 1)
+    return if (y == 0L) "OK" else "fail: $y"
+}
+
+// FILE: Nullable.java
+package android.support.annotation;
+
+public @interface Nullable {}
```

Comment:
If the issue is reproducible with our existing `org.jetbrains.annotations.Nullable`, I'd prefer to use it

Reply:
Done